### PR TITLE
feat: Implement claims

### PR DESCRIPTION
**Pull request by technosophos**
This adds the Claims datastructure as defined in CNAB section 400

Depends on #21

### CODE REVIEW

**Comment by glyn** [comment]
Please rebase.

File: src/cnab.rs [hunk 1]
```diff
@@ -20,39 +20,47 @@ pub struct Bundle {
...
+
     /// description is a short description of this bundle
     pub description: Option<String>,
     /// The list of images that comprise this bundle.
     ///
     /// Each image here is considered a constituent of the application described by this
     /// bundle.
-    pub images: Option<HashMap<String, Image>>,
-    /// inovcation_images is the list of available bootstrapping images for this bundle
+    pub images: Option<BTreeMap<String, Image>>,
```
Comment (glyn) [inline src/cnab.rs]: Just for my own understanding, was `BTreeMap` preferred over `HashMap` for predictable ordering or some other reason?

Comment (technosophos) [reply src/cnab.rs]: Predictable ordering.

**Comment by radu-matei** [comment]
I think this also needs a rebase?

File: src/claim.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    /// Extension space
+    pub custom: Option<serde_json::Value>,
+    /// Modification date
+    pub modified: DateTime<Utc>,
+    /// The name of the claim (e.g. the release name)
+    pub name: String,
+    /// Name/value pairs representing the outputs from the runtime
+    pub outputs: Option<BTreeMap<String, String>>,
+    /// Name/value paris that represent the parameter values
+    pub parameters: Option<BTreeMap<String, String>>,
```
Comment (radu-matei) [inline src/claim.rs]: When constructing a claim, are users expected to serialize the parameters before-hand?
Should this be a `Option<BTreeMap<String, Parameter>>`?

**Review by radu-matei — APPROVED** [review]

**Pull request merged**